Statistics/Distributions/Normal.cs

```diff
@@ -75,23 +94,35 @@ public IMessageLevels Validate(Utilities.IValidator<Normal> validator, out IEnum
         {
             return validator.IsValid(this, out msgs);
         }
-        private IRange<double> FiniteRange()
+        private IRange<double> FiniteRange(double min, double max)
         {
-            double min = double.NegativeInfinity, max = double.PositiveInfinity, p = 0, epsilon = 1 / 1000000000d;
+            double pmin = 0, epsilon = 1 / 1000000000d;
+            double pmax = 1 - pmin;
+            if (min.IsFinite() || max.IsFinite())//not entirely sure how inclusive or works with one sided truncation and the while loop below.
+            {
+                pmin = _Distribution.CumulativeDistribution(min);
+                pmax = _Distribution.CumulativeDistribution(max);
+            }
             while (!(min.IsFinite() && max.IsFinite()))
             {
-                p += epsilon;
-                if (!min.IsFinite()) min = _Distribution.InverseCumulativeDistribution(p);
-                if (!max.IsFinite()) max = _Distribution.InverseCumulativeDistribution(1 - p);
+                pmin += epsilon;
+                pmax -= epsilon;
+                if (!min.IsFinite()) min = _Distribution.InverseCumulativeDistribution(pmin);
+                if (!max.IsFinite()) max = _Distribution.InverseCumulativeDistribution(pmax);
             }
-            return IRangeFactory.Factory(p, 1 - p);
+            return IRangeFactory.Factory(pmin, pmax);
         }
         
         #region IDistribution Functions
         public double PDF(double x) => _Distribution.Density(x);
         public double CDF(double x) => _Distribution.CumulativeDistribution(x);
         public double InverseCDF(double p)
         {
+            if (Truncated && _Constructed)
+            {
+                //https://en.wikipedia.org/wiki/Truncated_normal_distribution
+                p = _ProbabilityRange.Min + (p) * (_ProbabilityRange.Max - _ProbabilityRange.Min);
```

Comment:
yay, wikipedia! is it worth adding this comment to all distribution classes?

Reply:
i think just for normal - it applies specifically to it - and im not certain i have it implemented right